### ISSUE TITLE
Many non-English word lists create non-word strings in place of some words.

### ISSUE THREAD

**Issue by leftdisconnected** [cmdwtf/KeePassDiceware#33]
KeePass v2.57
KPDiceware v1.60
Windows 10 22H2

### Issue:
Whether combined with other set or used alone, enabling the following word lists appears to include numeric or mixed strings in place of some words.  Some lists output a _lot_ of non-word strings while others only occasionally do:

Beale
Diceware
Finnish
French
Italian
Icelandic
Japanese
Polish
Spanish
Swedish

### Example:
Below are some 7-word examples from the Spanish word list taken from the Preview tab.  "-" is selected as a word separator and Salt is set to "no salt".  Note that the 7th word in the first example is "932" and so on:

Tarda-Roso-Europa-Museo-Talar-Grima-932
Tela-Ramal-4567-Labro-Trigo-124-E20
812-Gasto-Huaco-Gento-Fusor-Luso-W24
644-Bobo-Axial-Lijar-F88-Disco-Ojear
Laico-Roque-S72-Yunta-Na-221-Y25
C14-Zarbo-Belio-Frior-Capea-Yute-Ragua
F78-Olor-Tauro-317-817-Matiz-Comer
Sesmo-Deseo-Cuneta-Nitro-Diera-Sonda-P56
F30-Asuso-Ojal-Muer-Tocar-150-Lite
Visor-Borde-Firma-Doy-Grand-Roca-G20
Plan-Porco-Vuelta-Dupa-C47-Jeque-72
Greba-Osear-N23-Cuzco-Acoso-X67-Odre
Canon-Espina-Lampo-Hiato-Dheli-Na+-Ronce
Reble-Peto-Basto-Barba-K59-Virus-Melca

### Investigation:

1. I first assumed that this may be related to my language settings in Windows and/or KeePass as U.S. English editions of Windows have traditionally not included full language support, so I did further investigation targeting the Spanish word list.  I installed the "Spanish (Spain)" language in Windows and the Spanish language pack for KeePass.  I then set the Windows UI to Spanish and KeePass to Spanish and relogged the user session (required by Windows).

    Unfortunately, the issue persisted unchanged; enabling the Spanish word list resulted in passphrases like those shown above.  I also tried emptying the KeePass plugin cache and restarting in order to recompile KPDiceware, but no change.

2. My only other guess is that these strings somehow represent the random numbers used for lookup in the selected word list and the process is failing somehow.  That seems unlikely.

I apologize if I am overlooking something and will happily close the issue if this is user error.  Thank you.

**Comment by SecurityCze** [comment]
The issue is coming from the dictionary that is used inside the plugin.

For some reason it contains a large amount of "number" and "a letter followed by two numbers" rows. For example:

https://github.com/cmdwtf/KeePassDiceware/blob/72e82f433ff5de986e1ca60f5188ddf6bbcfa84f/Resources/Spanish.txt#L1-L11

https://github.com/cmdwtf/KeePassDiceware/blob/72e82f433ff5de986e1ca60f5188ddf6bbcfa84f/Resources/Spanish.txt#L1325-L1335

Only two fixes that come to mind:
- manually modify the current dictionary
- find some other Spanish dictionary

I took a quick look at other dictionaries:
- Finnish + French + Icelandic + Italian + Japanese + Norwegian + Polish + Swedish are most likely from the same source. They contain approximately 200 numbers and 30 symbols (at the file start) that could be simply deleted (if licence permits).
- Spanish is the worst offender. There is so much of random letter and number combinations. It could probably be quite easy to remove by simply deleting any row with a number.
- other dictionaries seem fine

**TLDR**: Definitely not user error. (Thank you for reporting) There are some expected lines 

---

The easiest might be to just:
```
for file in Resources/*; do
    sed -i '/[0-9]/d' "$file"
done
```
Which for every file in Resources directory removes any line that contains any number. That should be pretty safe.

```
printf "%-25s | %-8s | %-8s\n" "FILE" "ORIGINAL" "MODIFIED"
for file in Resources/*; do
    filename=`echo $file | cut -d'/' -f 2`
    original=`wc -l "$file" | cut -d' ' -f 1`
    modified=`wc -l "Resources-test/$filename" | cut -d' ' -f 1`
    printf "%-25s | %-8s | %-8s\n" "$filename" $original $modified
done
```

**Comment by nitz** [comment]
> simply deleted (if licence permits).

They certainly can. When I put this together, I simply inherited the lists from Strongbox since I was already pulling so much inspiration from it. Being a native English speaker, I hadn't even checked the other dictionaries as I wouldn't have known what to look for. 😅

I'm _guessing_ that they were there to push some of those lists up to ~7776 entries to line up with diceware's original list. That said, I don't think it's important that we need to maintain that sort of parity, since we're not rolling *actual* dice to pick words either.

I applied your quick fix there (at a54a0f6e3a93dba1b61c44b2b747879537735e04), @SecurityCze, but I did leave `Diceware.txt` and `SecureDrop.txt` unmodified, as I feel like it is in the spirit of the original diceware to keep those numeric entries. I also normalized the line endings you noticed on those 4 files!

**Comment by leftdisconnected** [comment]
Glad that this was a relatively simple issue and I apologize for not examining the word lists as it would have been obvious that these were actual entries.

Diceware's size gives us 12.92 bits per word (almost 13) and any word list _can_ be used alone, so I recommend that any included word list contain at least 5000 entries (specifically 4096) to provide a minimum of 12 bits per word.  I would prefer that each list either meet or exceed Diceware's 7776 entry count, but we depend upon others to create such lists and cannot place arbitrary demands upon entry count.  Enabling multiple lists at once makes it likely that we exceed original Diceware anyway.

It would seem useful to display information like "total words" and "bits per word" in the white space between salt sources and the word list window, but I understand that this will not happen.
